File: algos/solutions/a1twopointers/TS/palindrome/algo.test.ts
import { isPalindrome } from "./algo";

describe("Valid Palindrome - Two Pointers", () => {
    test("racecar should return true", () => {
        expect(isPalindrome("racecar")).toBe(true);
    });

    test("A man, a plan, a canal: Panama should return true", () => {
        expect(isPalindrome("A man, a plan, a canal: Panama")).toBe(true);
    });

    test("hello should return false", () => {
        expect(isPalindrome("hello")).toBe(false);
    });

    test("Empty string should return true", () => {
        expect(isPalindrome("")).toBe(true);
    });

    test("Spaces and punctuation should be ignored", () => {
        expect(isPalindrome("No 'x' in Nixon")).toBe(true);
    });

    test("Single character should return true", () => {
        expect(isPalindrome("x")).toBe(true);
    });
});

File: algos/solutions/a1twopointers/Go/palindrome/algo_test.go
package palindrome

import "testing"

func TestIsPalindrome(t *testing.T) {
	tests := []struct {
		input string
		want  bool
	}{
		{"racecar", true},
		{"Racecar", true},
		{"A man, a plan, a canal: Panama", true},
		{"hello", false},
		{"", true},
		{"   ", true},
		{"ABC", false},
		{"ABCBA", true},
		{"ABBA", true},
	}

	for _, tt := range tests {
		got := IsPalindrome(tt.input)
		if got != tt.want {
			t.Errorf("IsPalindrome(%q) = %v; want %v", tt.input, got, tt.want)
		}
	}
}
